coded_tools/agent_network_designer/create_agent_network_hocon.py

```diff
@@ -11,132 +11,17 @@
-    "#   - aaosa_call\n"
-    "#   - aaosa_command\n"
-    "#   - aaosa_instructions\n"
-    "# IMPORTANT:\n"
-    "# Ensure that you run `python -m run` from the top level of the repository.\n"
-    "# The path to this substitution file is **relative to the top-level directory**,\n"
-    "# so running the script from elsewhere may result in file not found errors.\n"
-    '    include "registries/aaosa.hocon"\n'
-    '    "llm_config": {\n'
-    '        "model_name": "gpt-4o",\n'
-    "    },\n"
-    '   "instructions_prefix": """\n'
-    "You are part of a team of assistants in "
-)
-HOCON_HEADER_REMAINDER = (
-    ".\n"
-    "Only answer inquiries that are directly within your area of expertise.\n"
-    "Do not try to help for other matters.\n"
-    "Do not mention what you can NOT do. Only mention what you can do.\n"
-    '""",\n'
-    '   "demo_mode": "You are part of a demo system, so when queried, make up a realistic response as if you '
-    'are actually grounded in real data or you are operating a real application API or microservice."\n'
-    '   "tools": [\n'
-)
-TOP_AGENT_TEMPLATE = (
-    "        {\n"
-    '            "name": "%s",\n'
-    '            "function": {\n'
-    '                "description": """\n'
-    "An assistant that answer inquiries from the user.\n"
-    '                """\n'
-    "            },\n"
-    '            "instructions": ${instructions_prefix} """\n'
-    "            Never express irrelevance unless you have first consulted all your tools.\n"
-    "            Once you have determined the relevant tools, do not express that to the user, rather,\n"
-    "            call all the relevant tools and make sure the command is fully serviced and express the end result.\n"
-    "%s\n"
-    '""" ${aaosa_instructions},\n'
-    '            "tools": [%s]\n'
-    "        },\n"
-)
-REGULAR_AGENT_TEMPLATE = (
-    "        {\n"
-    '            "name": "%s",\n'
-    '            "function": ${aaosa_call},\n'
-    '            "instructions": ${instructions_prefix} """\n'
-    "%s\n"
-    '""" ${aaosa_instructions},\n'
-    '            "command": ${aaosa_command},\n'
-    '            "tools": [%s]\n'
-    "        },\n"
-)
-LEAF_NODE_AGENT_TEMPLATE = (
-    "        {\n"
-    '            "name": "%s",\n'
-    '            "function": ${aaosa_call},\n'
-    '            "instructions": ${instructions_prefix} ${demo_mode} """\n'
-    "%s\n"
-    '""",\n'
-    "        },\n"
-)
-# pylint: disable=implicit-str-concat
-TOOLBOX_AGENT_TEMPLATE = "        {\n" '            "name": "%s",\n' '            "toolbox": "%s"\n' "        },\n"
-
-
-async def modify_registry(the_agent_network_hocon_str, the_agent_network_name):
-    """
-    Writes the agent network to a file and updates the manifest.hocon file.
-    :param the_agent_network_hocon_str: The agent network hocon string
-    :param the_agent_network_name: The file name, without the .hocon extension
-    """
```

Comment:
This modify_registry() method becomes the meat of FileSystemAgentNetworkPersistor.async_persist()